core-java/src/main/java/com/baeldung/abstractclasses/filereaders/BaseFileReader.java

```diff
@@ -1,19 +1,26 @@
 package com.baeldung.abstractclasses.filereaders;
 
 import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public abstract class BaseFileReader {
     
-    protected String filePath;
+    protected Path filePath;
     
-    protected BaseFileReader(String filePath) {
+    protected BaseFileReader(Path filePath) {
         this.filePath = filePath;
     }
     
-    public String getFilePath() {
+    public Path getFilePath() {
         return filePath;
     }
     
-    public abstract List<String> readFile() throws IOException;
+    public List<String> readFile() throws IOException {
+        return Files.lines(filePath).collect(Collectors.toList());

```

Comment:
@alejandrogervasio just because this is an intro article, I'm going to ask for just one more simplification. What would you think of this:

```java
return Files.lines(filePath).map(this::mapFileLine).collect(Collectors.toList());
```

And then

```java
protected abstract String mapFileLine(String line);
```

Reply:
@jzheaux Sound really good. I'll make the requested change today.

Reply:
@jzheaux I refactored the sample codes, so that they now reflect your suggestion. Definitively, this approach is better structured for a classic base abstract class/subclasses hierarchy. Nicely done!